infrap4d/infrap4d_main.cc

```diff
@@ -21,6 +28,12 @@ int main(int argc, char* argv[]) {
 #if defined(TOFINO_TARGET)
   return stratum::hal::tdi::TofinoMain(argc, argv).error_code();
 #elif defined(DPDK_TARGET)
+  gflags::ParseCommandLineFlags(&argc, &argv, false);
+  if (FLAGS_detach) {
+      daemonize_start(false);
+      daemonize_complete();
+  }
+
   krnlmon_init();
 
   krnlmon_shutdown();
```

Comment:
The function should end with `return 0;`.

I'm surprised we aren't getting a compiler warning about the function not returning a value. Or are we?

Reply:
I didn't notice any warning. Added return 0